src/sentry/hybridcloud/rpc/caching/service.py

```diff
@@ -162,7 +164,18 @@ def resolve_from(
             try:
                 metrics.incr("hybridcloud.caching.list.cached", tags={"base_key": self.base_key})
                 return [self.type_(**item) for item in json.loads(value)]
-            except (pydantic.ValidationError, JSONDecodeError, TypeError):
+            except (pydantic.ValidationError, JSONDecodeError, TypeError) as err:
+                metrics.incr(
+                    "hybridcloud.caching.list.failed_read",
+                    tags={
+                        "base_key": self.base_key,
+                        "err": type(err).__name__,
+                    },
+                )
+                if random.random() < 0.001:
```

Comment:
Put a really low sample rate on this as I don't think we need many examples to confirm my theory.